include/openthread/platform/radio.h

```diff
@@ -150,7 +150,9 @@ typedef uint16_t otShortAddress;
 OT_TOOL_PACKED_BEGIN
 struct otExtAddress
 {
-    uint8_t m8[OT_EXT_ADDRESS_SIZE]; ///< IEEE 802.15.4 Extended Address bytes
+    OT_TOOL_ALIGNED_FIELD_BEGIN(OT_EXT_ADDRESS_SIZE)
```

Comment:
Reading [EWARM](http://supp.iar.com/FilesPublic/UPDINFO/004916/arm/doc/EWARM_DevelopmentGuide.ENU.pdf), for `_Pragma("data_alignment")` I think this would request the struct filed to be aligned certain way but we actually want the whole `otExtAddress` to be packed. How would this work?

Reply:
The alignment of a plain struct is determined by the alignment of its first member.

However I am planning rather to approach things differently based on the discussion above. The otExtAddress struct does not need its member aligned, if the bytes are loaded explicitly from the array to form a 64-bit value. See https://blog.quarkslab.com/unaligned-accesses-in-cc-what-why-and-solutions-to-do-it-properly.html . I'll add commits to the pull request when I've made the changes.

Reply:
I think we want to keep the packed attribute because we do want unaligned access.

In the case of https://github.com/openthread/openthread/blob/150aa9b6db6f241bdc6056ff6c45c039cabc90b0/examples/platforms/cc1352/radio.c#L1580 we could simply use `memcpy()` or something equivalent.

Reply:
We use packed definition basically to allow unaligned access (being able to cast any pointer to the packed type). This is for example used in TLV parsing.

I would also echo @jwhui  suggestion to do copying of data/value and trying to avoid direct casting to `unit` pointer and deference.